### PR TITLE
Replace ArrayBuffer.transfer for ArrayBuffer.detach

### PR DESCRIPTION
**Pull request by Cellule**
Remove ArrayBuffer.transfer feature and replace it with a test function to only detach the ArrayBuffer.

---
This change is [Reviewable](https://reviewable.io/reviews/microsoft/chakracore/3621)

### CODE REVIEW

File: lib/Runtime/Library/ArrayBuffer.cpp [hunk 1]
```diff
@@ -429,22 +429,15 @@ namespace Js
...
-        if (args.Info.Count >= 3)
-        {
-            newBufferLength = ToIndex(args[2], JSERR_ArrayLengthConstructIncorrect, scriptContext, MaxArrayBufferLength);
-
-            // ToIndex above can call user script (valueOf) which can detach the buffer
-            if (arrayBuffer->IsDetached())
-            {
-                JavascriptError::ThrowTypeError(scriptContext, JSERR_DetachedTypedArray, _u("ArrayBuffer.transfer"));
-            }
+            JavascriptError::ThrowTypeError(scriptContext, JSERR_DetachedTypedArray, _u("ArrayBuffer.prototype.detach"));
```
Comment (akroshg) [inline lib/Runtime/Library/ArrayBuffer.cpp]: nit: this should be ArrayBuffer.detach as you have not added this method to prototype, you have added to constructor.

Comment (Cellule) [reply lib/Runtime/Library/ArrayBuffer.cpp]: Good catch. Initially I put it on the prototype and then decided to keep it similar to what ArrayBuffer.transfer had.

File: lib/Runtime/Library/ArrayBuffer.h [hunk 2]
```diff
@@ -142,15 +142,15 @@ namespace Js
...
+            static FunctionInfo Detach;
         };
 
         static Var NewInstance(RecyclableObject* function, CallInfo callInfo, ...);
         static Var EntrySlice(RecyclableObject* function, CallInfo callInfo, ...);
         static Var EntryIsView(RecyclableObject* function, CallInfo callInfo, ...);
         static Var EntryGetterByteLength(RecyclableObject* function, CallInfo callInfo, ...);
         static Var EntryGetterSymbolSpecies(RecyclableObject* function, CallInfo callInfo, ...);
-        static Var EntryTransfer(RecyclableObject* function, CallInfo callInfo, ...);
+        static Var EntryDetach(RecyclableObject* function, CallInfo callInfo, ...);
```
Comment (akroshg) [inline lib/Runtime/Library/ArrayBuffer.h]: nit: why not add this function in ENABLE_CONFIGOPTION only? same is true with EntryInfo::Detach

Comment (Cellule) [reply lib/Runtime/Library/ArrayBuffer.h]: I guess you're right. I was on the fence about it, but for no good reason.

**Comment by Penguinwizzard** [comment]
This change should fix #2733 (as it removes the broken test, as well as the code that was causing the leak).

**Pull request merged**